dev/2017/spring/alumni-chapters-and-clubs/alumni-chapters-and-clubs.js

```diff
@@ -0,0 +1,184 @@
+var map,
+    mapData,
+    center,
+    saControl,
+    saPosition,
+    $saMessage,
+    $saIcon,
+    isCentered,
+    $chapterText,
+    $clubText,
+    $memberText,
+    chapterCount,
+    clubCount,
+    memberCount,
+    chapterMarker,
+    clubMarker,
+    activeInfoWindows;
+
+var init = function() {
+  $chapterText = $('#chapters');
+  $clubText = $('#clubs');
+  $memberText = $('#members');
+  $saMessage = $('#sa-message');
+  $saIcon = $('#sa-icon');
+  activeInfoWindows = [];
+  chapterCount = clubCount = memberCount = 0;
+
+  if(window.google && google.maps) {
+    getData();
+  } else {
+    lazyLoadGoogleMap();
+  }
+};
+
+var lazyLoadGoogleMap = function() {
+  $.getScript('//maps.google.com/maps/api/js?sensor=false&callback=getData&key=AIzaSyBTQqIr2iqXIphR51yd3dCjudO4Z4PKYxM')
+    .fail(function(jqxhr, settings, ex) {
+      // do nothing  
+    });
+};
+
+var getData = function() {
+  var $container = $('#alumni-map'),
+      dataUrl = $container.data('map');
+  $.getJSON(dataUrl, function(data) {
+    mapData = data;
+    initializeMap();
+    initializeTables();
+  });
+};
+
+var initializeMap = function() {
+  center = new google.maps.LatLng(39.905227, -95.419666);
+  saPosition = new google.maps.LatLng(23.133774, 45.101088);
+  var mapOptions = {
+    center: center,
+    zoom: 4,
+    mapTypeId: google.maps.MapTypeId.SATELLITE,
+    scrollwheel: false
+  };
+  map = new google.maps.Map(document.getElementById('alumni-map'), mapOptions);
+
+  isCentered = true;
+  createControls();
+
+  chapterMarker = $('#alumni-map').data('chapter-marker');
```

Comment:
$('#alumni-map') could be cached in a global var and used across methods getData and initializeMap